fili-core/src/main/java/com/yahoo/bard/webservice/data/config/provider/MakerDictionary.java

```diff
@@ -0,0 +1,295 @@
+     *
+     * 1. The full package + class name specified in the className field
+     * 2. The name specified in the className field inside the MetricMaker package
+     * 3. If no className given, the Maker name specified (longSum), turned into a Maker class name
+     *    (LongSumMaker), inside the MetricMaker package
+     *
+     * @param suppliedMakerName The maker name the user specified
+     * @param suppliedClassName The class name the user provided
+     * @return The resolved class name
+     */
+    protected static String getClassName(String suppliedMakerName, String suppliedClassName) {
+        if (suppliedClassName == null || suppliedClassName.isEmpty()) {
+            return MAKER_PACKAGE_NAME + "." + suppliedMakerName.substring(0, 1)
+                    .toUpperCase(Locale.ENGLISH) + suppliedMakerName.substring(1) + "Maker";
+        } else if (suppliedClassName.contains(".")) {
+            return suppliedClassName;
+        } else {
+            return MAKER_PACKAGE_NAME + "." + suppliedClassName;
+        }
+    }
+
+    /**
+     * Construct a custom MetricMaker given its class name and arguments.
+     *
+     * @param name The pretty name ("longSum")
+     * @param className The class name for the maker
+     * @param args Arguments to the maker constructor
+     * @param dict Metric dictionary
+     * @return A metric maker
+     */
+    protected static MetricMaker buildCustomMaker(String name, String className, Object[] args, MetricDictionary dict) {
+        Class cls;
+        try {
+            cls = Class.forName(getClassName(name, className));
+        } catch (ClassNotFoundException e) {
+            throw new ConfigurationError("Error loading metric maker `" + name + "`: Could not find `" + className +
+                    "`" +
+                    " in classpath");
+        }
+
+        if (!MetricMaker.class.isAssignableFrom(cls)) {
+            throw new ConfigurationError("Error loading metric maker `" + name + "`: Could not instantiate " +
+                    className + "; seems to not inherit from MetricMaker");
+        } else {
+
+            Object[] instArgs = new Object[args.length + 1];
+            instArgs[0] = dict;
+            System.arraycopy(args, 0, instArgs, 1, args.length);
+
+            MetricMaker m = (MetricMaker) instantiateObjectFromArgs(cls, instArgs);
+
+            if (m == null) {
+                throw new ConfigurationError("Could not instantiate " + className + "; no suitable constructor found");
+            }
+
+            return m;
+        }
+    }
+
+    /**
+     * Instantiate an object from a class name and parameters.
+     *
+     * @param type the class to instantiate
+     * @param args the constructor arguments
+     * @param <T> the type of object to return
+     * @return an instantiated object, or null
+     */
+    public static <T> T instantiateObjectFromArgs(Class<? extends T> type, Object... args) {
+
```

Comment:
Cool/horrifying, there's a fine line :) But you're right, it could obviously construct any type of object.